### DISCUSSION TITLE
Only flush cache when delete() returns a count

### DISCUSSION THREAD

**Discussion by m-lotze** [GeneaLabs/laravel-model-caching#503] [Ideas]
The two methods `delete` and `forceDelete` in `Buildable` trait should imho only flush the cache, if the parent methods return an int > 0. Right now, the cache is flushed every time, even when the query did not actually deleted records from the database.

```
public function delete()
{
    $result = parent::delete();

    if ($result) {
        $this->cache($this->makeCacheTags())->flush();
    }

    return $result;
}
```

**Comment by mikebronner** [comment]
Thanks for the suggestion @m-lotze! This has been converted to a tracked issue:

➡️ **Issue #550:** https://github.com/mikebronner/laravel-model-caching/issues/550

Closing this discussion in favour of the issue.
